db/tables/operations/create.py

```diff
@@ -1,53 +1,51 @@
-from db.tables.operations.alter import comment_on_table
 
-
-def create_mathesar_table(name, schema, columns, engine, metadata=None, comment=None):
+def create_mathesar_table(engine, table_name, schema_oid, columns=[], constraints=[], comment=None):
     """
-    This method creates a Postgres table in the specified schema using the
-    given name and column list.  It adds internal mathesar columns to the
-    table.
+    Creates a table with a default id column.
+
+    Args:
+        engine: SQLAlchemy engine object for connecting.
+        table_name: Name of the table to be created.
+        schema_oid: The OID of the schema where the table will be created.
+        columns: The columns dict for the new table, in order. (optional)
+        constraints: The constraints dict for the new table. (optional)
+        comment: The comment for the new table. (optional)
+
+    Returns:
+        Returns the OID of the created table.
     """
-    columns = init_mathesar_table_column_list_with_defaults(columns)
-    create_schema(schema, engine, if_not_exists=True)
-    # We need this so that we can create multiple mathesar tables in the
-    # same MetaData, enabling them to reference each other in the
-    # SQLAlchemy context (e.g., for creating a ForeignKey relationship)
-    if metadata is None:
-        metadata = MetaData(bind=engine, schema=schema)
-    metadata.reflect()
-    # The exception raised by SQLAlchemy upon hitting a duplicate table in the
-    # schema is non-specific.
-    if (name, schema) in [(t.name, t.schema) for t in metadata.sorted_tables]:
-        raise DuplicateTable
-    table = Table(
-        name,
-        metadata,
-        *columns,
-        schema=schema
-    )
-    table.create(engine)
-    if comment is not None:  # this check avoids an unneeded DB call
-        comment_on_table(name, schema, engine, comment)
-    return table
-
-
-class DuplicateTable(Exception):
-    pass
-
-
-def create_string_column_table(name, schema, column_names, engine, comment=None):
+    return execute_msar_func_with_engine(
+        engine,
+        'add_mathesar_table',
+        schema_oid,
+        table_name,
+        json.dumps(columns),
+        json.dumps(constraints),
+        comment
+    ).fetchone()[0]
+
+
+def create_string_column_table(name, schema_oid, column_names, engine, comment=None):
     """
     This method creates a Postgres table in the specified schema, with all
     columns being String type.
     """
-    columns_ = [Column(name=column_name, type_=TEXT) for column_name in column_names]
-    table = create_mathesar_table(name, schema, columns_, engine, comment=comment)
+    columns_ = [
+        {
+            "name": column_name,
+            "type": {"name": PostgresType.TEXT.id}
+        } for column_name in column_names
+    ]
+    table_oid = create_mathesar_table(engine, name, schema_oid, columns_, comment=comment)
+    table = reflect_table_from_oid(table_oid, engine, metadata=get_empty_metadata())
```

Comment:
At first, I was going to request moving this logic up to the callers, but that would end up with it duplicated too much. Please add a TODO comment here specifying that we'd like to remove these and just return the OID at some point.

Reply:
> At first, I was going to request moving this logic up to the callers, but that would end up with it duplicated too much.

Yeah, that's why I added it here.